### PR TITLE
New port: Redwax modules - cryptographic support for apache

### PR DESCRIPTION
**Pull request by dirkx**
Set of new modules - done in one pull request as there are interdependencies.

Replaces #6587 

tested on
macOS 10.13, 10.14, 10.15
XCode 8 until 10.3.1

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)? 

yes - one for each port.

- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ra1nb0w** [comment]
as maintainers use this line
`maintainers  {@dirkx redwax.org:dirkx} openmaintainer`

Probably these new ports should be reviewed by apache2 maintainers: @ryandesign and @Schamschula

File: www/mod_ca/Portfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+depends_lib         port:apache2 port:pkgconfig
+
+use_configure       yes
+
+pre-build {
+    system "echo > ${worksrcpath}/mod_ca_ldap.c"
+    system "echo > ${worksrcpath}/mod_ca_ldap.h"
+}
+
```
Comment (reneeotten) [inline www/mod_ca/Portfile]: please remove newline at the end of the file

**Comment by dirkx** [comment]
@reneeotten is this fine as commit - or should break it into 8 commits; one for each module and then squash each ?

**Comment by reneeotten** [comment]
> @reneeotten is this fine as commit - or should break it into 8 commits; one for each module and then squash each ?

You should amend your previous commits to include these changes; in the end you'll want to have 1 commit for each port you add. The easiest way of doing is probably with ```git rebase -i HEAD~8```, then for each port one-by-one you change ```pick``` to ```edit```, make the required changes and do ```git commit -a --amend```. You finish the rebase by ```git rebase --continue``` and repeat this process until you're finished.

**Comment by dirkx** [comment]
Thanks - that was easily scriptable; will force push and see if it passes CI as well.

File: www/mod_ca/Portfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                mod_ca
+version             0.2.2
+categories          www security
+platforms           darwin
+maintainers         redwax.org:dirkx @dirkx openmaintainer
```
Comment (reneeotten) [inline www/mod_ca/Portfile]: this needs curly brackets to signify they belong to the same person, so like:
```{redwax.org:dirkx @dirkx} openmaintainer```

Comment (dirkx) [reply www/mod_ca/Portfile]: will do.

**Comment by dirkx** [comment]
all fixed & passes tests still. thanks

**Comment by dirkx** [comment]
@reneeotten, @Schamschula, @ryandesign: Changes to maintainer made; whitespace removed and Final CI test completed, is now ready for (re)review

**Comment by dirkx** [comment]
Right - no trouble doing that - but would that not lock it against macports version of OpenSSL?Whereas in some cases - it would be used against apples' openssl - which may be needed if you have hardware connections to things like chipcards ? Or is it better that the end user controls this with DYLIB environment set paths; and have macports be a self contained universe ?

**Comment by dirkx** [comment]
@reneeotten Made that change (after checking you can still run against OSX its openssl for HW tokens) - put the dependency on openssl also in the leafs - as they themselves also call openssl directly (or use structs from DER); so it is not just mod_ca which depends on openssl.

**Comment by dirkx** [comment]
Seems something timed out in he travis build (with no errors) - what is the right way to kick it off again ?

**Comment by raimue** [comment]
The pending test on Azure mac-10.13 actually finished successfully. Somehow it did not update the GitHub status, but there also does not seem to be any way to abort/restart the test. We will just have to ignore the result.

**Comment by reneeotten** [comment]
any comments @ryandesign or @Schamschula - if not, I will merge this tonight.

**Comment by reneeotten** [comment]
thanks @dirkx - in it goes!

**Comment by dirkx** [comment]
@reneeotten @Schamschula - thanks for all the help !

**Pull request merged**